app/src/main/java/nl/eduvpn/app/fragment/ConnectionStatusFragment.kt

```diff
@@ -38,23 +39,18 @@ import nl.eduvpn.app.viewmodel.BaseConnectionViewModel
 import nl.eduvpn.app.viewmodel.ConnectionStatusViewModel
 import java.util.Observable
 import java.util.Observer
-import javax.inject.Inject
 
 /**
  * The fragment which displays the status of the current connection.
  * Created by Daniel Zolnai on 2016-10-07.
  */
-class ConnectionStatusFragment : BaseFragment<FragmentConnectionStatusBinding>(), ConnectionInfoCallback {
+class ConnectionStatusFragment(private val vpnService: VPNService) : BaseFragment<FragmentConnectionStatusBinding>(), ConnectionInfoCallback {
```

Comment:
Supplying values via the constructor for fragments is very dangerous.
When the fragments gets recreated by the system, it will use the default no-argument constructor, and in that case the val will not be set anymore.
Can you revert to the Inject solution?

Reply:
Fixed in https://github.com/eduvpn/android/pull/318/commits/683e60176399a7a8fe8c805e6a35db97f98418e3